ConfigSpace/read_and_write/pcs_new.py

```diff
@@ -64,7 +64,9 @@
 pp_number = pp_e_notation | pp_float | pp_int
 pp_numberorname = pp_number | pp_param_name
 pp_log = pyparsing.Word("log")
-pp_connective = pyparsing.Word("||" + "&&")
+# A word matches each character as a set. So && is processed as &
+# https://pythonhosted.org/pyparsing/pyparsing.Word-class.html
+pp_connective = pyparsing.Word("|" + "&")
```

Comment:
Shouldn't this be `pyparsing.Literal` then?

Reply:
And the `log` above, too?

Reply:
Thanks for the catch of `log`. I have fixed that.

Leaving `pp_connective = pyparsing.Word("|" + "&")` as it is was intentional, as [this line](https://github.com/automl/ConfigSpace/blob/b2fe51e7f03cc080640093d30bbaaa76a11a3b59/ConfigSpace/read_and_write/pcs_new.py#L80) assumes you will be checking for `&` or `|` not `&|`.

I do not know much about pyparsing, but probably we can do some `Combine` or `And` to create this behaviour. In other words, if one just makes this literal [this](https://github.com/franchuterivera/ConfigSpace/blob/63b781a031d6768566001932b0749f82df6caf6d/test/read_and_write/test_pcs_converter.py#L492) unit test fails.

Let me know how you want to proceed (that is re-writing the pyparsing statement or keeping the `pyparsing.Word("|" + "&")`).

Reply:
That makes sense. Can you check whether there's a union of literals we could use here to rewrite this statement to check for either `"??"` or `"||"`?

Reply:
The change looks good. There's no need to keep this simple, this should be used only very rarely anyways, so it can be slow...